src/utils/IntlWrapper.ts

```diff
@@ -66,69 +46,47 @@ export class IntlDuration {
     return this;
   }
 
-  setRelativeTime(relativeTime: number, convertFrom?: keyof typeof dateUnitToMs) {
-    this.relativeTime = Number(relativeTime);
-    if (convertFrom) this.toTimestamp(convertFrom);
-    return this;
-  }
-
-  setDuration(duration: durationFormat) {
+  // For {hours: 2, minutes: 15}
+  setDuration(duration: durationFormat, style: durationStyle = 'Duration') {
     this.duration = duration;
+    this.process(style);
     return this;
   }
 
-  setDurationFormatStyle(style: durationFormatStyle) {
-    this.durationFormatStyle = style;
+  // For {minutes: 155}
+  setDurationFormatted(duration: durationFormat, style: durationStyle = 'Duration') {
+    const relativeTime = IntlDuration.durationToMs(duration);
+    this.setRelativeTime(relativeTime, 'milliseconds', style);
     return this;
   }
 
-  setDurationStyle(style: durationStyle) {
-    this.durationStyle = style;
+  // For 9000 (seconds) (relative unit - so don't need to convert)
+  setRelativeTime(
+    relativeTime: number,
+    convertFrom: keyof typeof dateUnitToMs,
+    style: durationStyle = 'Duration',
+  ) {
+    const duration = IntlDuration.relativeToDuration(relativeTime, convertFrom);
+    this.setDuration(duration, style);
     return this;
   }
 
-  protected checkForNow(): boolean {
-    if (!this.duration) return false;
-    const keys = Object.keys(this.duration);
-    for (let i = 0; i < keys.length; i++) {
-      const key = keys[i];
-      const value = this.duration[key];
-      if (value !== 0 && key !== 'seconds') return false;
-      if (value <= 30 && key === 'seconds') break;
-    }
-    return true;
-  }
-
-  protected transform() {
-    switch (this.durationStyle) {
-      case 'Duration':
-        if (this.relativeTime)
-          ({ time: this.duration, isFuture: this.isFuture } = timestampToTime(this.relativeTime));
-        break;
-      case 'Progress':
-        if (this.relativeTime)
-          ({ time: this.duration, isFuture: this.isFuture } = timestampToTime(this.relativeTime));
-        break;
-      case 'M/H/D/Y':
-        if (this.relativeTime)
-          ({ time: this.duration, isFuture: this.isFuture } = timestampToTime(this.relativeTime));
-        break;
-      default:
-        break;
-    }
+  // For 1733320467580 (date timestamp) (absolute unit - need to convert to relative)
+  setTimestamp(
+    timestamp: number,
+    style: durationStyle = 'Duration',
+    relativeTo: Date | number = new Date(),
+  ) {
+    const duration = IntlDuration.timestampToDuration(timestamp, relativeTo);
+    this.setDuration(duration, style);
     return this;
   }
 
-  protected format() {
-    switch (this.durationStyle) {
-      case 'Duration':
-        break;
+  protected process(style: durationStyle) {
```

Comment:
How about adding additional param to `relativeToDuration` for `convertFrom` so it will look like `IntlDuration.relativeToDuration(WhatToConvert, WhichToConvertFrom, WhatToConvertInto)` or we rely on the fact that user will specify unit to convert from as the first one like `IntlDuration.relativeToDuration(9000, ['seconds', 'minutes', 'hours', 'days', 'years']);` - 'seconds' is input here

Reply:
To keep everything as simple as possible I would use ms as early as possible and have a different function just for the timeIntervalToMs

Reply:
Done it